components/AttackSlot.jsx

```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+
+export const AttackSlot = ({ register, index, attackRemove }) => {
+  return (
+    <>
+      <StatWrapper>
+        <Stat>
+          <LabelText htmlFor={`attackSlotArray.${index}.attack-name`}>
```

Comment:
In general I'd think about using camelCase instead of kebab-case for all of your field names. That way it feels a bit more JavaScripty™️.